src/java.desktop/unix/classes/sun/awt/X11/XWindowPeer.java

```diff
@@ -694,12 +694,34 @@ public void checkIfOnNewScreen(Rectangle newBounds) {
                         // Completely on this screen - done!
                         newScreenNum = i;
                         newGC = gds[i].getDefaultConfiguration();
+                        if (target != null
+                                && target.getBackground() != SystemColor.window
```

Comment:
I agree. I recently worked on a bug involving window backgrounds and translucency is only one aspect of it. The color itself may be different and this conditional looks like it would definitely trigger in the case where the window changes color between screens.